include/mbedtls/ecdh.h

```diff
@@ -40,6 +40,39 @@
 
 #include "mbedtls/ecp.h"
 
+/**
```

Comment:
I don't think that should be a doxygen comment. This is now internal, so there's not need to this to be included in the public API documentation. I suggest removing the second `*` here, and the `\` directives below (or `check-doxy-blocks.pl` will complain).

Also, we should add an explicit warning that this in internal, something to the effect of "don't look at this unless you're a compiler" - ideally we would hide this from the user, but unfortunately if needs to be in the header file as it affects the definition of the context structure.